nsxt/resource_nsxt_manager_cluster.go

```diff
@@ -0,0 +1,173 @@
+				Sensitive:   true,
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Description: "The username for login",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceNsxtManagerClusterCreate(d *schema.ResourceData, m interface{}) error {
+	nodes := getStringListFromSchemaList(d, "nodes")
+	if len(nodes) == 0 {
+		return fmt.Errorf("Please provide at least one node ip")
+	} else if len(nodes) == 1 {
+		log.Printf("[INFO] Only one node ip provided, skip joining nodes process")
+		return resourceNsxtManagerClusterRead(d, m)
+	}
+	masterNodeIP := nodes[0]
+	clusterID, certSha256Thumbprint, err := getClusterInfoFromMasterNode(masterNodeIP, d, m)
+	if err != nil {
+		return err
+	}
+	workerNodeIPs := nodes[1:]
+	for _, workerNodeIP := range workerNodeIPs {
+		err := joinNodeToCluster(clusterID, certSha256Thumbprint, workerNodeIP, masterNodeIP, d, m)
+		if err != nil {
+			return err
+		}
+	}
+	d.SetId(clusterID)
+	d.Set("cluster_id", clusterID)
+	return resourceNsxtManagerClusterRead(d, m)
+}
+
+func getClusterInfoFromMasterNode(ip string, d *schema.ResourceData, m interface{}) (string, string, error) {
+	connector := getPolicyConnector(m)
+	client := nsx.NewClusterClient(connector)
+	clusterConfig, err := client.Get()
+	if err != nil {
+		return "", "", fmt.Errorf("Failed to get cluster info from node %s: %s", ip, err)
+	}
+	clusterID := *clusterConfig.ClusterId
+	nodes := clusterConfig.Nodes
+	node := nodes[0]
+	apiListenAddr := node.ApiListenAddr
+	certSha256Thumbprint := *apiListenAddr.CertificateSha256Thumbprint
+	return clusterID, certSha256Thumbprint, nil
+}
+
+func getOverrideNsxtClient(host string, d *schema.ResourceData, clients interface{}) interface{} {
+	// The API of joining a new node to cluster needs to be called with the new node's ip,
+	// to do that we initialize a new nsxtClient with new node's ip as its host
+	c := clients.(nsxtClients)
+	newClients := nsxtClients(c)
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+	if !strings.HasPrefix(host, "https://") {
+		host = fmt.Sprintf("https://%s", host)
+	}
+	newClients.Host = host
+	newClients.NsxtClientConfig.UserName = username
+	newClients.NsxtClientConfig.Password = password
+	newClients.NsxtClientConfig.Host = host
+	apiClient := (*api.APIClient)(newClients.NsxtClient)
+	apiClientConfig := apiClient.GetCfg()
+	apiClientConfig.UserName = username
+	apiClientConfig.Password = password
+	apiClientConfig.Host = host
+	newAPIClient, err := api.NewAPIClient(apiClientConfig)
+	if err != nil {
+		log.Fatal(err)
+		return nil
+	}
+	newClients.NsxtClient = newAPIClient
+	return newClients
+}
+
+func joinNodeToCluster(clusterID string, certSha256Thumbprint string, workerNodeIP string, masterNodeIP string, d *schema.ResourceData, m interface{}) error {
+	newNsxClients := getOverrideNsxtClient(workerNodeIP, d, m).(nsxtClients)
+	connector := getPolicyConnector(newNsxClients)
+	client := nsx.NewClusterClient(connector)
```

Comment:
This connector would connect to master node from provider config, and I think here we're supposed to connect to the node API endpoint?